### PR TITLE
install google cloud sdk (gsutil)

### PR DESCRIPTION
**Pull request by ssusie**
adding google cloud sdk; in particular gsutil to work with `--debug-dump-gcs` in xpk.

### CODE REVIEW

File: maxtext_dependencies.Dockerfile [hunk 1]
```diff
@@ -7,6 +7,13 @@ RUN apt-get update && apt-get install -y curl gnupg git numactl
...
 
+# Add the Google Cloud SDK package repository
+RUN echo "deb [signed-by=/usr/share/keyrings/cloud.google.gpg] https://packages.cloud.google.com/apt cloud-sdk main" | tee -a /etc/apt/sources.list.d/google-cloud-sdk.list
+RUN curl https://packages.cloud.google.com/apt/doc/apt-key.gpg | apt-key --keyring /usr/share/keyrings/cloud.google.gpg add -
+
+# Install the Google Cloud SDK
+RUN apt-get update && apt-get install -y google-cloud-sdk
+
 # Install the Google Cloud SDK
 RUN curl -sSL https://sdk.cloud.google.com | bash
```
Comment (Obliviour) [inline maxtext_dependencies.Dockerfile:18]: Hmm fishy that we install it twice. Why did we change to curl (breaking gsutil initially). I wonder if we need curl for anothwr reason?

Comment (rwitten) [reply maxtext_dependencies.Dockerfile:18]: Yes super suspicious Can we get rid of the curl?

(And I imagine you've already tested this??)

Comment (rwitten) [reply maxtext_dependencies.Dockerfile:18]: Finally can you add a test that just runs locally and makes sure gsutil is installed in our docker image? Just a unit test that does something like `gsutil --version`. (The problem is that some important flows depends on gsutil but most don't so right now we discover it being broken super slowly.)

**Review by rwitten — CHANGES_REQUESTED** [review]

File: maxtext_dependencies.Dockerfile (hunk 1 above)
Comment (ssusie) [reply maxtext_dependencies.Dockerfile]: Addressed comments. Can you PTAL?

**Review by rwitten — APPROVED** [review]

**Pull request merged**